### PR TITLE
Widgets user guide.

### PR DESCRIPTION
**Pull request by rohitvarkey**
This guide covers

- Introduction to how widgets work
- Sliders
- Buttons
- Checkboxes
- Togglebuttons
- Radiogroups
- Codemirror

The `datepicker` asset isn`t correct, so I've commented that out for the time being. Should I install [this](https://github.com/bendavis78/paper-date-picker/) as a replacement?

@shashi @alinchis Please review and give me your suggestions!

### CODE REVIEW

**Comment by alinchis** [comment]
+ I like that they are interactive, even modify code and re-render
- the Signal value in the Slider example is frozen, the same in the codemirror
- the initial value of the Signal can be set to be different from the widget. as a result the first render may not be accurate, like the slider second example and the radio button. Can you use the value of the signal to set the initial widget value?
+ adding `datepicker` is a good idea. I believe it will make Escher more attractive to users, the more ready-to-use elements it has the better

**Comment by rohitvarkey** [comment]
> the Signal value in the Slider example is frozen, the same in the codemirror

They work fine for me. 

> the initial value of the Signal can be set to be different from the widget. as a result the first render may not be accurate, like the slider second example and the radio button. Can you use the value of the signal to set the initial widget value?

I've set the initial value of the Signal and the widget to be same, so it should be correct in the first render itself!

File: examples/user-guide/widgets.jl [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        They are well suited to be used for users to input numerical inputs, such
+        as the number of iterations for a simulation to run. Sliders can be created
+        using the `slider` function and providing a range as the input. The starting
+        value can also be provided using the `value` keyword argument.
+        """,
+        listing(
+            """
+            # Create a Signal
+            inp = Signal(5.0)
+            # Creating a slider and subscribing it to the signal
```
Comment (shashi) [inline examples/user-guide/widgets.jl]: good to use the same tense everywhere

File: examples/user-guide/widgets.jl [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        listing(
+            """
+            code = Signal("1+1")
+            codewidget = codemirror("1+1") >>> code
+            map(code) do code
+                vbox(
+                    codewidget,
+                    eval(parse(code))
+                )
+            end
```
Comment (shashi) [inline examples/user-guide/widgets.jl]: this is cool!

File: examples/user-guide/widgets.jl [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    push!(window.assets, "date")
+    push!(window.assets, "codemirror")
+    push!(window.assets, "tex")
+    colors = ["violet", "blue", "green", "yellow", "orange", "red"]
+    colortile(c, w, h) = fillcolor(c, size(w, h, empty))
+
+    vbox(
+        title(3, "Widgets"),
+        md"""Escher provides a set of input widgets that can be used to create
+        dynamic UI's that take inputs from users and are interactive. Widgets
```
Comment (shashi) [inline examples/user-guide/widgets.jl]: *UIs

**Comment by shashi** [comment]
+1

is the goal to explain `foldp` in a different guide? also forms, samplers (e.g. code mirror with ctrl+enter) and collectors (minesweeper-like things)

**Comment by rohitvarkey** [comment]
I was using http://escher-jl.org/widgets-api.html as a reference. I think this can focus on just introducing widgets with their default behaviours. 

Having a separate guide for explaining Reactive features (`foldp` etc) and then one more on forms, samplers and collectors will be better than putting everything in one guide I feel.

**Comment by shashi** [comment]
There is also the opinionated http://escher-jl.org/reactive-guide.html

**Comment by alinchis** [comment]
you are right, I was messing with it and got those issues. on reload, they worked as expected.

**Comment by rohitvarkey** [comment]
I've been trying to get the date picker to work, but when included in an Escher session, the layout is all messed up. Funnily enough, it becomes correct when the browser is resized! 
screen shot 2016-06-25 at 12 12 29 pm

I'm wondering if we should remove the datepicker code for now and merge this PR.

**Comment by alinchis** [comment]
I believe we should avoid confusion.
the users might think something is wrong with their code. 
on the other hand if you issue a usage warning, it might be accetable :)

**Comment by randyzwitch** [comment]
Is there anything keeping this from being merged? There are two discussions going on here, improving the docs and tangent code issues

**Comment by shashi** [comment]
I think @rohitvarkey never got around to dealing with the ugly calendar issue. I say we merge this for now.

**Comment by rohitvarkey** [comment]
Sorry everyone. Lost track of the PR for a bit there. I couldn't get the calendar to work nicely and from what I remember, didn't find any good replacements either. So the `datepicker` code is essentially dead code as @shashi mentioned. We should probably remove it.

**Comment by shashi** [comment]
Yes we should remove it.

**Pull request merged**